### PR TITLE
Add support for abstract classes

### PR DESCRIPTION
**Pull request by sbrunk**
Including abstract fields and methods.

### CODE REVIEW

File: src/main/scala/org/scalajs/tools/tsimporter/parser/TSDefParser.scala [hunk 1]
```diff
@@ -297,7 +297,8 @@ class TSDefParser extends StdTokenParsers with ImplicitConversions {
     | "public" ^^^ Modifier.Public
     | "readonly" ^^^ Modifier.ReadOnly
     | "protected" ^^^ Modifier.Protected
-  )
+    | lexical.Identifier("abstract") ^^^ Modifier.Abstract
+    )
```
Comment (sjrd) [inline src/main/scala/org/scalajs/tools/tsimporter/parser/TSDefParser.scala]: The formatting for the `)` was broken here (there should be only 2 spaces before it).

File: src/main/scala/org/scalajs/tools/tsimporter/sc/Printer.scala [hunk 2]
```diff
@@ -139,7 +139,8 @@ class Printer(private val output: PrintWriter, outputPackage: String) {
           if (sym.modifiers(Modifier.Const)) "val"
           else if (sym.modifiers(Modifier.ReadOnly)) "def"
           else "var"
-        pln"  $access$decl $name: ${sym.tpe} = js.native"
+        p"  $access$decl $name: ${sym.tpe}"
+        pln"${if (sym.modifiers(Modifier.Abstract)) "" else " = js.native"}"
```
Comment (sjrd) [inline src/main/scala/org/scalajs/tools/tsimporter/sc/Printer.scala]: IMO the following would be more readable:
```scala
if (!sym.modifiers(Modifier.Abstract))
  p" = js.native"
pln""
```

File: src/main/scala/org/scalajs/tools/tsimporter/sc/Printer.scala [hunk 3]
```diff
@@ -158,7 +159,8 @@ class Printer(private val output: PrintWriter, outputPackage: String) {
           p"  ${modifiers}def $name"
           if (!sym.tparams.isEmpty)
             p"[${sym.tparams}]"
-          pln"($params): ${sym.resultType} = js.native"
+          p"($params): ${sym.resultType}"
+          pln"${if (sym.modifiers(Modifier.Abstract)) "" else " = js.native"}"
```
Comment (sjrd) [inline src/main/scala/org/scalajs/tools/tsimporter/sc/Printer.scala]: Same readability comment here.

**Comment by sjrd** [comment]
This should now be rebased on master, and the `.ts` files be renamed to `.d.ts` to adapt to the changes done in #56.

**Comment by sbrunk** [comment]
Rebased and nitpicks should have been addressed, too :)

**Pull request merged**